orchestrator/graphql/__init__.py

```diff
@@ -0,0 +1,110 @@
+from httpx import HTTPStatusError
+from starlette.requests import Request
+from strawberry.fastapi import GraphQLRouter
+from strawberry.http import GraphQLHTTPResponse
+from strawberry.types import ExecutionContext, ExecutionResult
+from strawberry.utils.logging import StrawberryLogger
+
+from orchestrator.graphql.extensions.deprecation_checker_extension import make_deprecation_checker_extension
+from orchestrator.graphql.extensions.ErrorCollectorExtension import ErrorCollectorExtension
+from orchestrator.graphql.resolvers.process import resolve_processes
+from orchestrator.graphql.resolvers.product import resolve_products
+from orchestrator.graphql.schemas.process import Process
+from orchestrator.graphql.schemas.product import Product
+from orchestrator.graphql.types import CustomContext
+from orchestrator.security import get_oidc_user, get_opa_security_graphql
+from orchestrator.settings import app_settings
+
+api_router = APIRouter()
+
+logger = structlog.get_logger(__name__)
+
+
+@strawberry.type(description="Orchestrator queries")
+class Query:
+    processes: list[Process] = strawberry.field(resolve_processes, description="Returns list of processes")
+    products: list[Product] = strawberry.field(resolve_products, description="Returns list of products")
+
+
+class PythiaGraphqlRouter(GraphQLRouter):
+    def __init__(self, schema: strawberry.federation.Schema, **kwargs: Any) -> None:
+        super().__init__(schema, **kwargs)
+
+    @staticmethod
+    def _format_graphql_error(error: GraphQLError) -> GraphQLFormattedError:
+        if isinstance(error.original_error, HTTPStatusError):
+            error.extensions["http_status_code"] = {  # type: ignore
+                f"{error.original_error.request.url}": error.original_error.response.status_code
+            }
+        return format_error(error)
+
+    async def process_result(self, request: Request, result: ExecutionResult) -> GraphQLHTTPResponse:
+        data: GraphQLHTTPResponse = {"data": result.data}
+
+        if result.errors:
+            data["errors"] = [self._format_graphql_error(error) for error in result.errors]
+
+        return data
+
+
+class CustomSchema(strawberry.federation.Schema):
+    def process_errors(
+        self,
+        errors: list[GraphQLError],
+        execution_context: ExecutionContext | None = None,
+    ) -> None:
+        """Override error processing to reduce verbosity of the logging.
+
+        https://strawberry.rocks/docs/types/schema#handling-execution-errors
+        """
+        for error in errors:
+            if (
+                isinstance(error.original_error, HTTPStatusError)
+                and error.original_error.response.status_code == HTTPStatus.NOT_FOUND
+            ):
+                message = str(error.original_error).splitlines()[0]  # Strip "For more info"
+                StrawberryLogger.logger.debug(message)
+            else:
+                StrawberryLogger.error(error, execution_context)
```

Comment:
I think we can drop this; in our internal graphql project this had a usecase because of the many other REST APIs it queried. But orchestrator's graphql won't query anything other than a SQL (and perhaps someday a NoSQL) database

Reply:
I do think we want to keep the `CustomSchema`, so it can be extended instead of the strawberry `Schema`? (at least renaming it to `OrchestratorSchema`)